packages/devtools_app/lib/src/standalone_ui/ide_shared/property_editor/property_editor_inputs.dart

```diff
@@ -124,24 +124,66 @@ class _DropdownInputState<T> extends State<_DropdownInput<T>>
         padding: denseSpacing,
       ),
       isExpanded: true,
+      selectedItemBuilder: (context) {
+        return widget.property.propertyOptions.map((option) {
+          return DropdownMenuItem(
+            value: option.text,
+            child: _DropdownContent(option: option, showDefaultLabel: false),
+          );
+        }).toList();
+      },
       items:
           widget.property.propertyOptions.map((option) {
             return DropdownMenuItem(
-              value: option,
-              child: Text(option, style: theme.fixedFontStyle),
+              value: option.text,
+              child: _DropdownContent(option: option, showDefaultLabel: true),
             );
           }).toList(),
       onChanged: (newValue) async {
-        await editProperty(
-          widget.property,
-          valueAsString: newValue,
-          controller: widget.controller,
-        );
+        if (newValue != widget.property.valueDisplay) {
+          await editProperty(
+            widget.property,
+            valueAsString: newValue,
+            controller: widget.controller,
+          );
+        }
       },
     );
   }
 }
 
+class _DropdownContent extends StatelessWidget {
+  const _DropdownContent({
+    required this.option,
+    required this.showDefaultLabel,
+  });
+
+  final PropertyOption option;
+  final bool showDefaultLabel;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        Expanded(
+          child: Text(
+            option.text,
+            style: Theme.of(context).fixedFontStyle,
+            overflow: TextOverflow.ellipsis,
+          ),
+        ),
+        if (showDefaultLabel && option.isDefault) ...[
+          const Spacer(),
```

Comment:
nit: we could also achieve this effect with `mainAxisAlignment: MainAxisAlignment.spaceBetween` and then could get rid of this widget and therefore the list + spread operator too

Reply:
Good point, done!